lib/ansible/cli/__init__.py

```diff
@@ -305,15 +310,14 @@ def setup_vault_secrets(loader, vault_ids, vault_password_files=None,
         if last_exception and not found_vault_secret:
             raise last_exception
 
+        if initialize_context:
+            VaultSecretsContext.initialize(VaultSecretsContext(vault_secrets))
+
         return vault_secrets
 
     @staticmethod
-    def _get_secret(prompt):
-
-        secret = getpass.getpass(prompt=prompt)
-        if secret:
-            secret = to_unsafe_text(secret)
-        return secret
+    def _get_secret(prompt: str) -> str:
+        return NotATemplate().tag(getpass.getpass(prompt=prompt))
```

Comment:
would this not be the default? do we need to mark things as non templates?

Reply:
The `NotATemplate` feature has been removed.